.travis.yml

```diff
@@ -1,106 +1,38 @@
-sudo: false
+# We will use docker to set up our environment, so don't use any particular
+# language in Travis itself
+language: generic
 
-language: cpp
+# Enable docker
+sudo: required
+services:
+    - docker
 
-os:
-  # - linux
-  - osx
+# Disable automatic submodule fetching ...
+git:
+    submodules: false
 
-osx_image: xcode8
+# ... in order to ensure we do a non-shallow submodule fetch
+before_install:
+    - git fetch --unshallow --tags
```

Comment:
IIRC we were getting some nasty issues w.r.t. either commits or tags not being available.  I can try removing this once the `beaver` fix lands.